model/insights.go

```diff
@@ -260,6 +261,23 @@ func StartOfDayForTimeRange(timeRange string, location *time.Location) *time.Tim
 	return &resultTime
 }
 
+// GetStartOfDayForTimeRange gets the unix start time in milliseconds from the given time range.
```

Comment:
I know it'd be a big change for me to request. Could we please rename this function to something like `StartOfDayForTimeRangeWithError`? Open to correction, but this might be confusing later on.

Reply:
I think the idea is that once all three PRs are merged, we can get rid of the old function, so it won't be confusing. What do you think?

Reply:
I agree, but 0/5 on the suggestion :) resolving this for now. good work!